src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,387 @@
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetails(taskId: string): Promise<any> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/tasks/${encodeURIComponent(taskId)}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        const etag: string | undefined = response['@odata.etag'];
+
+        if (!etag) {
+          return Promise.reject(`Error fetching task details`);
+        }
+
+        return Promise.resolve(etag);
+      });
+  }
+
+  private async generateUserAssignments(args: CommandArgs, logger: Logger) {
+    const assignments: { [userId: string]: any; } = {};
+
+    if (args.options.assignedToUserNames) {
+      const userIds = await this.getUserIds(logger, args.options.assignedToUserNames);
+      userIds.map(x => assignments[x] = {
+        "@odata.type": "microsoft.graph.plannerAssignment", "orderHint": " !"
+      });
+    }
+
+    if (args.options.assignedToUserIds) {
+      args.options.assignedToUserIds.split(',').map(x => assignments[x] = {
+        "@odata.type": "microsoft.graph.plannerAssignment", "orderHint": " !"
+      });
+    }
+    return assignments;
+  }
+
+  private getBucketId(args: CommandArgs, planId: string): Promise<string> {
+    if (args.options.bucketId) {
+      return Promise.resolve(args.options.bucketId);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { id: string; name: string; }[] }>(requestOptions)
+      .then((response) => {
+        const bucket: { id: string; name: string; } | undefined = response.value.find(val => val.name === args.options.bucketName);
+    
+        if (!bucket) {
+          return Promise.reject(`The specified bucket does not exist`);
+        }
+    
+        return Promise.resolve(bucket.id);
+      });
+  }
+
+  private getPlanId(args: CommandArgs): Promise<string> {
+    if (args.options.planId) {
+      return Promise.resolve(args.options.planId);
+    }
+
+    return this
+      .getGroupId(args)
+      .then((groupId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans?$filter=(owner eq '${groupId}')`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: { id: string; title: string; }[] }>(requestOptions);
+      }).then((response) => {
+        const plan: { id: string; title: string; } | undefined = response.value.find(val => val.title === args.options.planName);
```

Comment:
Instead of defining a custom type `{ id: string; title: string; }`, let's use the `Plan` type from MS Graph typings